algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -98,7 +102,10 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
 
   KOKKOS_FUNCTION
   iterator_type& operator--() {
-    --m_current_index;
+    if constexpr (is_contiguous)
+      m_data--;
+    else
+      m_data -= m_stride;
```

Comment:
Any chance you can make the `m_stride` be `1` if it's contiguous, to keep the code readable ?
```suggestion
  iterator_type& operator--() {
      m_data -= m_stride;
```

Reply:
Probably worth checking that `stride_0` is compile-time or not.